### PR TITLE
Get curretContext when host-cluster-context is empty

### PR DESCRIPTION
**Pull request by qiujian16**
**What this PR does / why we need it**:

Get current context in kubeconfig when host-cluster-context
is not set during kubefedctl join

**Which issue(s) this PR fixes**

Fixes https://github.com/kubernetes-sigs/kubefed/issues/1064

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/kubefedctl/join.go [hunk 1]
```diff
@@ -184,6 +184,17 @@ func (j *joinFederation) Run(cmdOut io.Writer, config util.FedConfig) error {
 		return err
 	}
 
+	// Set HostClusterContext as current context in kube config if it is not set
+	if j.HostClusterContext == "" {
+		clientConfig := config.GetClientConfig(j.HostClusterContext, j.Kubeconfig)
```
Comment (marun) [inline pkg/kubefedctl/join.go]: The call on this line is identical to the one on line 168 which suggests calling `hostConfig.RawConfig()`.

Comment (marun) [reply pkg/kubefedctl/join.go]: ~~The call on this line is functionally identical to the one on line 168 which suggests calling `hostConfig.RawConfig()`.~~

Edit: I missed that `HostConfig` calls `GetClientConfig` internally and they return different types. Please ensure that `GetClientConfig` is only called once e.g.

```golang
hostClientConfig := ...
hostConfig, err := hostClientConfig.ClientConfig()
...
if j.HostClusterContext ... {
  rawConfig, err := ...
}
```

Comment (qiujian16) [reply pkg/kubefedctl/join.go]: done, thanks

**Comment by xunpan** [comment]
/lgtm

The CI failed. I just restart it to have another try.

**Comment by marun** [comment]
Thank you! 

/approve

**Pull request merged**